libnavui-shield/src/test/java/com/mapbox/navigation/ui/shield/RoadShieldContentManagerImplTest.kt

```diff
@@ -0,0 +1,441 @@
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.junit.Assert
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class RoadShieldContentManagerImplTest {
+
+    @get:Rule
+    var coroutineRule = MainCoroutineRule()
+
+    @Test
+    fun `request waits for all results to be available (success and manual cancellation), async`() =
+        coroutineRule.runBlockingTest {
+            val cache = mockk<ShieldResultCache>()
+            val contentManager = createContentManager(cache)
+
+            val legacyShieldUrl = "url_legacy"
+            val toDownloadLegacy = mockk<RouteShieldToDownload.MapboxLegacy> {
+                every { url } returns legacyShieldUrl
+            }
+            val expectedLegacyShield = RouteShield.MapboxLegacyShield(
+                legacyShieldUrl,
+                byteArrayOf()
+            )
+            val expectedLegacyResult = RouteShieldResult(
+                expectedLegacyShield,
+                RouteShieldOrigin(
+                    isFallback = false,
+                    originalUrl = legacyShieldUrl,
+                    originalErrorMessage = ""
+                )
+            )
+            coEvery {
+                cache.getOrRequest(toDownloadLegacy)
+            } coAnswers {
+                delay(500L)
+                ExpectedFactory.createValue(expectedLegacyShield)
+            }
+
+            val designShieldUrl = "url"
+            val toDownloadDesign = mockk<RouteShieldToDownload.MapboxDesign> {
+                every { url } returns designShieldUrl
+                every { legacyFallback } returns null
+            }
+            val expectedDesignResult = RouteShieldError(
+                designShieldUrl,
+                RoadShieldContentManagerImpl.CANCELED_MESSAGE
+            )
+            coEvery {
+                cache.getOrRequest(toDownloadDesign)
+            } coAnswers {
+                try {
+                    delay(1000L)
+                    ExpectedFactory.createError("error")
+                } catch (ex: CancellationException) {
+                    ExpectedFactory.createError(RoadShieldContentManagerImpl.CANCELED_MESSAGE)
+                }
+            }
+
+            var result: List<Expected<RouteShieldError, RouteShieldResult>>? = null
+            pauseDispatcher {
+                launch {
+                    result = contentManager.getShields(listOf(toDownloadLegacy, toDownloadDesign))
+                }
+                // run coroutines until they hit the download delays
+                runCurrent()
+                // advance by enough to only download one shield
+                advanceTimeBy(600L)
+                contentManager.cancelAll()
+            }
+            Assert.assertEquals(expectedLegacyResult, result!![0].value)
```

Comment:
```suggestion
           assertEquals(expectedLegacyResult, result!![0].value)
```

This applies to other tests below 👇

Reply:
Resolved.